sdk/monitor/monitor-query/TROUBLESHOOTING.md

```diff
@@ -0,0 +1,377 @@
+
+### Server timeouts when executing logs query request
+
+Some complex Kusto queries can take a long time to complete and such queries are aborted by the
+service if they run for more than 3 minutes. For such scenarios, the query APIs on `LogsQueryClient`, provide options to
+configure the timeout on the server. The server timeout can be extended up to 10 minutes.
+
+You may see an error as follows:
+
+```json
+{
+  "name": "RestError",
+  "code": "GatewayTimeout",
+  "statusCode": 504,
+  "request": {
+    "url": "https://api.loganalytics.io/v1/workspaces/<workspace_id>/query",
+    "headers": {
+      "content-type": "application/json",
+      "accept": "application/json",
+      "prefer": "REDACTED",
+      "accept-encoding": "gzip,deflate",
+      "user-agent": "azsdk-js-monitor-query/1.0.1 azsdk-js-monitor-log-query/1.0.1 core-rest-pipeline/1.7.0 Node/v14.16.1 OS/(x64-Windows_NT-10.0.19044)",
+      "x-ms-client-request-id": "b3269277-18d7-4971-9eb8-4a2259bc8576",
+      "authorization": "REDACTED",
+      "content-length": "76"
+    },
+    "method": "POST",
+    "timeout": 0,
+    "disableKeepAlive": false,
+    "streamResponseStatusCodes": {},
+    "withCredentials": false,
+    "requestId": "b3269277-18d7-4971-9eb8-4a2259bc8576",
+    "allowInsecureConnection": false,
+    "enableBrowserStreams": false
+  },
+  "details": {
+    "error": {
+      "code": "GatewayTimeout",
+      "message": "Gateway timeout",
+      "innerError": {
+        "code": "GatewayTimeout",
+        "message": "Kusto query timed out"
+      },
+      "correlationId": "2572d63e-99ae-4d3c-92d0-032ba7651ed3"
+    }
+  },
+  "message": "Gateway timeout"
+}
+
+```
+
+The following code shows a sample of setting the server timeout to 10 minutes. By setting this server timeout, the Azure Monitor Query library automatically extends the client timeout to wait for 10 minutes for the server to respond. You don't need to configure your HTTP client to extend the response timeout, as shown in the previous section.
+
+```ts
+import { DefaultAzureCredential } from "@azure/identity";
+import { LogsQueryClient } from "@azure/monitor-query";
+
+const credential = new DefaultAzureCredential();
+const logsQueryClient = new LogsQueryClient(credential);
+const result = await logsQueryClient.queryWorkspace(
+        monitorWorkspaceId,
+        kustoQuery,
+        { duration: Durations.oneHour },
+        {serverTimeoutInSeconds: 600}
+        );
+```
+
+### Partially successful logs query requests
+
+By default, if the execution of a Kusto query resulted in a partially successful response, the Azure Monitor Query
+client library will return the result of type `LogsQueryPartialResult` with the `status` field of `result` object set to `PartialFailure` to indicate to the user that the query was not fully successful. In case of multiple queries, when the results from all queries aren't successful, the `status` field of `result` object may be set to `Failure`.
```

Comment:
```suggestion
By default, if the execution of a Kusto query resulted in a partially successful response, the Azure Monitor Query client library will return the result of type `LogsQueryPartialResult` with the `status` field of `result` object set to `PartialFailure` to indicate to the user that the query was not fully successful. In case of multiple queries, when the results from all queries aren't successful, the `status` field of `result` object may be set to `Failure`.
```